rs/nervous_system/integration_tests/src/pocket_ic_helpers.rs

```diff
@@ -2764,9 +2837,18 @@ pub mod sns {
             swap_canister_id: PrincipalId,
             swap_parameters: SwapParameters,
         ) {
-            for (i, amount) in swap_direct_participations(swap_parameters)
-                .iter()
-                .enumerate()
+            for (i, amount) in swap_direct_participations(
```

Comment:
Maybe this `swap_direct_participations` function should take the entire `swap_parameters`

Reply:
One use case for this function is to generate participation amounts for the swap that already had some direct participations (e.g. user participated via NNS dapp web UI), so passing `swap_parameters` is not viable.

Perhaps `remaining_swap_participations` is a more suitable name